### ISSUE TITLE
用PaddleSeg中的模型导出文件导出模型和参数后，在用之前部署的C++预测库时遇到问题

### ISSUE THREAD

**Issue by eddieheyutong** [PaddlePaddle/models#3582]
如题，我在通过export_model.py文件导出了下载的Paddle 官方预训练模型，deeplabv3p_xception65_bn_cityscapes，然后用之前部署的C++预测库进行预测的时候，报出以下错误
/root/projects/PaddleSeg/inference/build# ./demo --conf=/root/projects/PaddleSeg/inference/conf/deploy.yaml --input_dir=/root/projects/PaddleSeg/inference/images/cityscapes/
WARNING: Logging before InitGoogleLogging() is written to STDERR
I1015 16:15:59.258088 19235 analysis_predictor.cc:88] Profiler is deactivated, and no profiling report will be generated.
I1015 16:15:59.278381 19235 op_compatible_info.cc:201] The default operator required version is missing. Please update the model version.
I1015 16:15:59.278416 19235 analysis_predictor.cc:841] MODEL VERSION: 0.0.0
I1015 16:15:59.278431 19235 analysis_predictor.cc:843] PREDICTOR VERSION: 0.0.0
W1015 16:15:59.278589 19235 analysis_predictor.cc:855]  - Version incompatible (1) batch_norm
W1015 16:15:59.278600 19235 analysis_predictor.cc:855]  - Version incompatible (1) bilinear_interp
W1015 16:15:59.278607 19235 analysis_predictor.cc:855]  - Version incompatible (1) concat
W1015 16:15:59.278614 19235 analysis_predictor.cc:855]  - Version incompatible (1) conv2d
W1015 16:15:59.278622 19235 analysis_predictor.cc:855]  - Version incompatible (1) depthwise_conv2d
W1015 16:15:59.278630 19235 analysis_predictor.cc:855]  - Version incompatible (1) dropout
W1015 16:15:59.278637 19235 analysis_predictor.cc:855]  - Version incompatible (1) elementwise_add
W1015 16:15:59.278646 19235 analysis_predictor.cc:855]  - Version incompatible (1) feed
W1015 16:15:59.278652 19235 analysis_predictor.cc:855]  - Version incompatible (1) fetch
W1015 16:15:59.278659 19235 analysis_predictor.cc:855]  - Version incompatible (1) reduce_mean
W1015 16:15:59.278667 19235 analysis_predictor.cc:855]  - Version incompatible (1) relu
W1015 16:15:59.278674 19235 analysis_predictor.cc:855]  - Version incompatible (1) scale
W1015 16:15:59.278682 19235 analysis_predictor.cc:855]  - Version incompatible (1) softmax
W1015 16:15:59.278688 19235 analysis_predictor.cc:855]  - Version incompatible (1) transpose2
W1015 16:15:59.278705 19235 analysis_predictor.cc:148] WARNING: Results may be DIFF! Using same versions between model and lib.
--- Running analysis [ir_graph_build_pass]
--- Running analysis [ir_graph_clean_pass]
--- Running analysis [ir_analysis_pass]
--- Running IR pass [is_test_pass]
--- Running IR pass [simplify_with_basic_ops_pass]
--- Running IR pass [fc_fuse_pass]
--- Running IR pass [fc_elementwise_layernorm_fuse_pass]
--- Running IR pass [conv_affine_channel_fuse_pass]
--- Running IR pass [conv_eltwiseadd_affine_channel_fuse_pass]
--- Running IR pass [conv_bn_fuse_pass]
---  detected 78 subgraphs
--- Running IR pass [conv_eltwiseadd_bn_fuse_pass]
--- Running IR pass [conv_elementwise_add_act_fuse_pass]
---  detected 53 subgraphs
--- Running IR pass [conv_elementwise_add2_act_fuse_pass]
--- Running IR pass [conv_elementwise_add_fuse_pass]
---  detected 26 subgraphs
--- Running IR pass [transpose_flatten_concat_fuse_pass]
--- Running IR pass [runtime_context_cache_pass]
--- Running analysis [ir_params_sync_among_devices_pass]
I1015 16:16:00.083369 19235 ir_params_sync_among_devices_pass.cc:41] Sync params from CPU to GPU
--- Running analysis [adjust_cudnn_workspace_size_pass]
--- Running analysis [inference_op_replace_pass]
--- Running analysis [ir_graph_to_program_pass]
I1015 16:16:00.203075 19235 analysis_predictor.cc:473] ======= optimize end =======
---  skip [feed], feed -> image
---  skip [save_infer_model/scale_0], fetch -> fetch
W1015 16:16:00.567735 19235 device_context.cc:235] Please NOTE: device: 0, CUDA Capability: 61, Driver API Version: 10.1, Runtime API Version: 10.0
W1015 16:16:00.571164 19235 device_context.cc:243] device: 0, cuDNN Version: 7.6.
runtime = 14236
size of outputs[0]: (1,19,512,512,)
E1015 16:16:01.581390 19235 seg_predictor.cpp:74]  [FATAL] unequal: input vs output [7864320|4980736]
runtime = 10389
size of outputs[0]: (1,19,512,512,)
E1015 16:16:01.682529 19235 seg_predictor.cpp:74]  [FATAL] unequal: input vs output [7864320|4980736]
runtime = 8927
size of outputs[0]: (1,19,512,512,)
E1015 16:16:01.771071 19235 seg_predictor.cpp:74]  [FATAL] unequal: input vs output [7864320|4980736]
runtime = 9973
size of outputs[0]: (1,19,512,512,)
E1015 16:16:01.857481 19235 seg_predictor.cpp:74]  [FATAL] unequal: input vs output [7864320|4980736]
runtime = 6773
size of outputs[0]: (1,19,512,512,)
E1015 16:16:01.934250 19235 seg_predictor.cpp:74]  [FATAL] unequal: input vs output [7864320|4980736]
runtime = 6660
size of outputs[0]: (1,19,512,512,)
E1015 16:16:02.018646 19235 seg_predictor.cpp:74]  [FATAL] unequal: input vs output [7864320|4980736]

然后图片的文件夹中并没有生成预测的图片，想请问下大佬们这个是什么原因呢？

**Comment by willthefrog** [comment]
已assign相关同学跟进。

**Comment by sjtubinlong** [comment]
@eddieheyutong 您好，这个错误是模型预测输出的尺寸和配置文件中的`eval_crop_width`, `eval_crop_height` 不一致引起的。

从日志中可以看到模型输出的尺寸是`512 x 512`的

```
size of outputs[0]: (1,19,512,512,)
```
但是
需要预测库配置的尺寸是 `1025 x 2049`的，请检查一下。
